src/ray/core_worker/test/core_worker_test.cc

```diff
@@ -823,6 +827,21 @@ TEST_F(SingleNodeTest, TestObjectInterface) {
     ASSERT_EQ(*results[i]->GetMetadata(), *buffers[i].GetMetadata());
   }
 
+  // Test Get() returns early when it encounters an error.
+  std::vector<ObjectID> ids_with_exception(ids.begin(), ids.end());
+  ids_with_exception.push_back(ObjectID::FromRandom());
+  std::vector<RayObject> buffers_with_exception(buffers.begin(), buffers.end());
+  std::string error_string = std::to_string(ray::rpc::TASK_EXECUTION_EXCEPTION);
+  char error_buffer[error_string.size()];
+  size_t len = error_string.copy(error_buffer, error_string.size(), 0);
+  buffers_with_exception.emplace_back(
```

Comment:
isn't the above just a very intricate way of doing `error_string.data()`?

Reply:
Unfortunately not, we need a non-const reference to the data. We could also do `const_cast` if you'd prefer.

Reply:
ah no, let's not do that. Probably a const version of the constructor would be in order https://github.com/ray-project/ray/blob/d8f580469096b02fc1ac66a2a90cd2b0ef9dafba/src/ray/common/buffer.h#L51 and an immutable version of the buffer, let's leave as is for this PR.